api/circulation_exceptions.py

```diff
@@ -46,6 +46,12 @@ class AuthorizationFailedException(CirculationException):
 class PatronAuthorizationFailedException(AuthorizationFailedException):
     status_code = 400
 
+class PatronCreationFailedException(CirculationException):
+    status_code = 500
+
```

Comment:
I would call this something like `RemotePatronCreationFailedException` so it doesn't look like the problem is we couldn't create the patron's record _in the ILS_.